execution_engine/src/core/engine_state/genesis.rs

```diff
@@ -1152,33 +1389,64 @@ where
         Ok(standard_payment_hash)
     }
 
-    fn create_accounts(&self, total_supply_key: Key) -> Result<(), GenesisError> {
+    fn is_private_chain(&self) -> bool {
+        self.exec_config.administrative_accounts().next().is_some()
+    }
+
+    pub(crate) fn create_accounts(
+        &self,
+        total_supply_key: Key,
+        payment_purse_uref: URef,
+    ) -> Result<(), GenesisError> {
         let accounts = {
-            let mut ret: Vec<GenesisAccount> = self.exec_config.accounts().to_vec();
+            let mut ret: Vec<GenesisAccount> = self.exec_config.accounts().cloned().collect();
             let system_account = GenesisAccount::system();
-            ret.push(system_account); // todo load bearing remove or not? probably dont need anymore
+            ret.push(system_account);
             ret
         };
 
+        if self.is_private_chain()
+            && self
+                .exec_config
+                .administrative_accounts()
+                .duplicates_by(|admin| &admin.public_key)
+                .next()
+                .is_some()
+        {
+            // Ensure no duplicate administrator accounts are specified as this might raise errors
+            // during genesis process when administrator accounts are added to associated keys.
+            return Err(GenesisError::DuplicatedAdministratorEntry);
+        }
+
+        let is_private_chain = self.exec_config.administrative_accounts().next().is_some();
+
         let mut total_supply = U512::zero();
 
-        for account in accounts {
+        for account in accounts.iter() {
             let account_hash = account.account_hash();
-            let main_purse = self.create_purse(account.balance().value())?;
+            let main_purse = match account {
+                GenesisAccount::System if self.is_private_chain() => payment_purse_uref,
+                _ => self.create_purse(account.balance().value())?,
+            };
+            total_supply += account.balance().value();
 
             let key = Key::Account(account_hash);
-            let stored_value = StoredValue::Account(Account::create(
-                account_hash,
-                Default::default(),
-                main_purse,
-            ));
-
-            let _ = self.tracking_copy.borrow_mut().write(key, stored_value);
 
-            total_supply += account.balance().value();
+            let account_config = if is_private_chain {
```

Comment:
You can remove [line 1421](https://github.com/casper-network/casper-node/pull/3029/files#diff-35b4940898fafa984f49d3047d48f7c7fddeab728b173e247bcde8857fa56af6R1421) and then use

```suggestion
            let account_config = if self.is_private_chain() {
```

Reply:
`account_config` is removed, it was unnecessary